### PR TITLE
Fixes ENYO-527

### PR DESCRIPTION
**Pull request by ryanjduffy**
## Issue
Need to provide API to determine which controls are visible for a DataList and DataGridList.

## Fix
Add `getVisibleControlRange()` to enyo.DataList which returns the `start` and `end` index of the visible controls. For a control to be considered visible, a certain percentage (established by the `visibleThreshold` property) must be visible.

Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

### CODE REVIEW

**Comment by seunghoh** [comment]
I can see the problem on garnet is fixed. It works well on garnet DataList

File: source/ui/data/VerticalDelegate.js [hunk 1]
```diff
@@ -128,6 +128,8 @@
 			// get our initial sizing cached now since we should actually have
 			// bounds at this point
 			this.updateBounds(list);
+			// calc offset of pages to scroller client
+			this.calcScrollOffset(list);
```
Comment (aarontam) [inline source/ui/data/VerticalDelegate.js]: Do we need to add this call to VerticalGridDelegate's `rendered` as well?

Comment (ryanjduffy) [inline source/ui/data/VerticalDelegate.js]: yes

Comment (ryanjduffy) [inline source/ui/data/VerticalDelegate.js]: Done!

Comment (aarontam) [inline source/ui/data/VerticalDelegate.js]: :+1:

**Pull request merged**